daemon/image_delete.go

```diff
@@ -132,15 +146,16 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 				if first {
 					return err
 				}
-
 			}
-
 		}
 	}
 	return nil
 }
 
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
+	directlyUsedBy := []string{}
```

Comment:
Because we want to allow the user to untag the image with transitive links, I think dividing the two link types is much easier.